File: backend/src/main/java/isep/webapp/terasync/model/user/DeliveryPerson.java
package isep.webapp.terasync.model.user;

import isep.webapp.terasync.model.odSystem.CustomerOrder;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.util.List;

@Entity
@Setter
@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class DeliveryPerson extends User {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "deliveryPersonId")
    private int deliveryPersonId;
    @Column(name = "firstName")
    private String firstName;
    @Column(name = "lastName")
    private String lastName;
    @Column(name = "email")
    private String email;
    @Column(name = "username")
    private String username;
    @Column(name = "password")
    private String password;
    @Column(name = "token")
    private String token;
    @CreationTimestamp
    @Column(name = "registrationDate")
    private String registrationDate;
    @Column(name = "deliveryServiceId")
    private int deliveryServiceId;
    @Column(name = "emailVerified")
    private boolean emailVerified;
    @Column(name = "pfpImgPath")
    private String pfpImgPath;

    @OneToMany(
            fetch = FetchType.EAGER,
            targetEntity = CustomerOrder.class,
            cascade = CascadeType.ALL
    )
    @JoinColumn(name = "deliveryPersonId", updatable = false)
    private List<CustomerOrder> customerOrders;

    @Override
    public int getUserId() {
        return deliveryPersonId;
    }

    @Override
    public String getName() {
        return firstName + " " + lastName;
    }

    @Override
    public boolean getEmailVerified() {
        return emailVerified;
    }
}

File: front/src/app/model/user/delivery.person.ts
import {User} from "./user";
import {CustomerOrder} from "../odSystem/customer.order";
import {DeliveryService} from "./delivery.service";

export class DeliveryPerson extends User {
  override deliveryPersonId: number | undefined;
  override firstName: string;
  override lastName: string;
  override deliveryServiceId: number;

  override customerOrders: CustomerOrder[] = [];

  constructor(firstName: string, lastName: string, email: string,
              username: string, password: string, deliveryServiceId: number,
              deliveryPersonId?: number | undefined,
              registrationDate?: string | undefined, token?: string | undefined,
              emailVerified?: boolean | undefined, pfpImgPath?: string | undefined, name?: string | undefined) {
    super(email, username, password, deliveryPersonId, registrationDate, token, emailVerified, pfpImgPath, name);
    this.deliveryPersonId = deliveryPersonId;
    this.firstName = firstName;
    this.lastName = lastName;
    this.deliveryServiceId = deliveryServiceId;
  }

  static override fromJson(jsonDeliveryPerson: DeliveryPerson) {
    let deliveryPerson = new DeliveryPerson(jsonDeliveryPerson.firstName, jsonDeliveryPerson.lastName, jsonDeliveryPerson.email,
      jsonDeliveryPerson.username, jsonDeliveryPerson.password, jsonDeliveryPerson.deliveryServiceId,
      jsonDeliveryPerson.deliveryPersonId, jsonDeliveryPerson.registrationDate, jsonDeliveryPerson.token,
      jsonDeliveryPerson.emailVerified, jsonDeliveryPerson.pfpImgPath, jsonDeliveryPerson.name)

    deliveryPerson.customerOrders = CustomerOrder.initializeCustomerOrders(jsonDeliveryPerson);

    return deliveryPerson;
  }

  static initializeDeliveryPeople(jsonDeliveryService: DeliveryService) {
    let deliveryPeople: DeliveryPerson[] = [];

    if (jsonDeliveryService.deliveryPeople !== undefined) {
      for (let jsonDeliveryPerson of jsonDeliveryService.deliveryPeople) {
        deliveryPeople.push(DeliveryPerson.fromJson(jsonDeliveryPerson));
      }
    }

    return deliveryPeople;
  }
}
